### DISCUSSION TITLE
macOS non-native fullscreen window is restored as regular window

### DISCUSSION THREAD

**Discussion by tysonmote** [ghostty-org/ghostty#3765] [Feature Requests, Ideas] [answered]
When I quit Ghostty and re-open it, my non-native fullscreen window is restored as a regular, non-fullscreen window. Instead, it should restore the window as a non-native fullscreen window.

## Reproduction

Use this config:

```
fullscreen = true
macos-non-native-fullscreen = true
```

Open Ghostty with a non-native fullscreen window. Quit and then re-open Ghostty. The window will now be a regular, non-fullscreen window instead of a non-native fullscreen window.

## My setup

- macOS 15.2 (24C101)
- Ghostty 1.0.0 (config below)

<img width="412" alt="image" src="https://github.com/user-attachments/assets/34477a2e-0b3c-40c8-803b-8400649bd2d9" />

**Comment by nounder** [comment]
Experiencing the same issue.

**Comment by mitchellh** [comment, marked as answer]
#8435 finally confirmed
